### PR TITLE
feat: Basic agent table component

### PR DESCRIPTION
**Pull request by iamsivin**
**Screenshot**

**Pagination not enabled**
image

**Pagination enabled**
image

**Sorting enabled**
image

**Loom video**
https://www.loom.com/share/87270d5c95fd4579b75e5158015d8684?sid=e26dd87f-845a-4ed3-ad4b-f8042174113e

### CODE REVIEW

File: src/components/Table/Table.story.vue [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+import Table from './Table.vue';
+import AgentSection from './components/AgentSection.vue';
+
+const agentData = [
+  {
+    id: 1,
+    name: 'John Doe',
+    email: 'john@gmail.com',
+    thumbnail: 'https://randomuser.me/api/portraits/men/49.jpg',
+    open_conversations: 3,
```
Comment (pranavrajs) [inline src/components/Table/Table.story.vue]: Let us not use underscore.

File: src/components/Table/Table.vue [hunk 2]
```diff
@@ -0,0 +1,50 @@
+<script setup>
+import { ref } from 'vue';
+
+const props = defineProps({
+  data: Array,
+  columns: Array,
+});
+
+import { useVueTable, FlexRender, getCoreRowModel } from '@tanstack/vue-table';
```
Comment (pranavrajs) [inline src/components/Table/Table.vue]: I would keep imports at the top.

File: src/components/Table/Table.story.vue [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+  {
+    accessorKey: 'unattended_conversations',
+    header: 'Unattended',
+  },
+];
+</script>
+
+<template>
+  <Story title="Table/AgentTable" width="100%">
+    <Variant title="AgentTable">
```
Comment (pranavrajs) [inline src/components/Table/Table.story.vue]:
```suggestion
    <Variant title="User list">
```

File: src/components/Table/Table.story.vue [hunk 4]
```diff
@@ -0,0 +1,116 @@
...
+  },
+  {
+    accessorKey: 'unattended_conversations',
+    header: 'Unattended',
+  },
+];
+</script>
+
+<template>
+  <Story title="Table/AgentTable" width="100%">
```
Comment (pranavrajs) [inline src/components/Table/Table.story.vue]:
```suggestion
  <Story title="Table/UserList" width="100%">
```

File: src/components/Table/Table.vue [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+            class="p-3 text-sm text-slate-1200 border-t border-solid border-slate-400"
+          >
+            <FlexRender :render="cell.column.columnDef.cell" :props="cell.getContext()" />
+          </td>
+        </tr>
+      </tbody>
+    </table>
+  </div>
+</template>
+<style></style>
```
Comment (pranavrajs) [inline src/components/Table/Table.vue]: Unnecessary style

File: src/components/Table/components/AgentSection.vue [hunk 6]
```diff
@@ -0,0 +1,28 @@
+<script setup>
+defineProps({
+  name: String,
+  email: String,
+  thumbnail: String,
+});
+</script>
+<template>
```
Comment (pranavrajs) [inline src/components/Table/components/AgentSection.vue]: I would rename this file something more generic like UserCard or something of that sort.

**Review by pranavrajs — CHANGES_REQUESTED** [review]
Minor comments, otherwise looks good.

**Comment by pranavrajs** [comment]
@iamsivin Can you please include pagination as well into the component? We don't need to handle those in the dashboard code separately.

**Comment by iamsivin** [comment]
> @iamsivin Can you please include pagination as well into the component? We don't need to handle those in the dashboard code separately.

Sure will add it.

**Comment by iamsivin** [comment]
> @iamsivin Can you please include pagination as well into the component? We don't need to handle those in the dashboard code separately.

@pranavrajs Done https://github.com/chatwoot/design/pull/11/commits/f8eb38afaae72ee02786e066f9c157424bd21f94

File: src/components/Table/components/UserCard.vue [hunk 7]
```diff
@@ -0,0 +1,28 @@
...
+  thumbnail: String,
+});
+</script>
+<template>
+  <div class="flex items-center gap-3 w-full">
+    <img
+      v-if="thumbnail"
+      class="w-10 h-10 rounded-full border border-solid border-slate-400"
+      :src="thumbnail"
+      alt="Profile picture"
```
Comment (pranavrajs) [inline src/components/Table/components/UserCard.vue]: Let us use "name + profile picture"

Comment (iamsivin) [reply src/components/Table/components/UserCard.vue]: @pranavrajs Done https://github.com/chatwoot/design/pull/11/commits/1c3e3344ee744d644ac2cb776e13d554a5daf3ca

**Review by pranavrajs — APPROVED** [review]

**Pull request merged**